### PR TITLE
basic R examples with 2.0 CLI

### PR DESCRIPTION
**Pull request by danielsc**
# PR into Azure/azureml-examples

## Checklist

I have:

- [ ] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

File: cli/jobs/train/r/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- R
+- azurecli
+products:
+- azure-machine-learning
+description: Learn how to submit R jobs to an AzureMLCompute Cluster using the AzureML CLI v2.
+---
+
+# R using the AzureML CLI v2
+
+This directory provides some sample projects to run R using the AzureML CLI v2.
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/README.md:13]: "2.0 CLI"

File: cli/jobs/train/r/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## Environment
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/README.md:15]: add newline after H2

File: cli/jobs/train/r/accident-prediction/accident_r_job.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  local_path: ./src
+command: >-
+  Rscript accidents.R
+  --data {inputs.training_data}
+environment:
+  docker:
+    build:
+      dockerfile: file:Dockerfile
+compute:
+  target: azureml:goazurego
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/accident-prediction/accident_r_job.yml:13]: not a compute target in the workspace, see `setup.sh` (you probably want `cpu-cluster`)

File: cli/jobs/train/r/accident-prediction/accident_r_job.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+inputs:
+  training_data:
+    data: 
+      local_path: ./data
+    mode: mount
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/accident-prediction/accident_r_job.yml:18]: add a description to this job (for the table of examples)

File: cli/jobs/train/r/basic-train-model/basic_r_job.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+      dockerfile: file:Dockerfile
+compute:
+  target: azureml:goazurego
+code:
+  local_path: ./src
+inputs:
+  iris:
+    data: 
+      local_path: ./data
+    mode: mount
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/basic-train-model/basic_r_job.yml:16]: add description

File: cli/jobs/train/r/basic-train-model/basic_r_job.yml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandJob.schema.json
+experiment_name: r-basic-example
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/basic-train-model/basic_r_job.yml:2]: follow standard order of job sections (see contributing guidelines) - experiment_name should be toward bottom

File: cli/jobs/train/r/basic-train-model/data/iris.csv [hunk 7]
```diff
@@ -0,0 +1,151 @@
+"Sepal.Length","Sepal.Width","Petal.Length","Petal.Width","Species"
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/basic-train-model/data/iris.csv:1]: don't put data in the repo, use cloud data - see other examples for public iris url

**Review by lostmygithubaccount — CHANGES_REQUESTED** [review]

File: cli/jobs/train/r/basic-train-model/data/iris.csv (hunk 7 above)
Comment (danielsc) [reply cli/jobs/train/r/basic-train-model/data/iris.csv:1]: couldn't get the file mounting to work -- also, showing how to use local data matches more the users' use cases (rather than public internet data).

Comment (lostmygithubaccount) [reply cli/jobs/train/r/basic-train-model/data/iris.csv:1]: Azure users and the enterprise ML professionals we target are expected to have their data in the cloud - the only difference is the authentication, which we do need to have a good story and samples around. I don't think leading users in the direction of uploading data from local for runs in the cloud is a good pattern

File: cli/jobs/train/r/README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
+---
+page_type: sample
+languages:
+- R
+- azurecli
+products:
+- azure-machine-learning
+description: Learn how to submit R jobs to an AzureMLCompute Cluster using the AzureML 2.0 CLI.
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/README.md:8]: submit -> create

AzureML -> Azure ML

I would avoid saying "AzureMLCompute Cluster", that seems like an undefined feature/noun in the product

File: cli/jobs/train/r/README.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+---
+
+# R using the AzureML 2.0 CLI
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/README.md:11]: AzureML -> Azure ML

maybe reword to "Train machine learning models with R using the 2.0 CLI for Azure ML"?

File: cli/jobs/train/r/README.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+This directory provides some sample projects to run R using the AzureML 2.0 CLI.
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/README.md:13]: AzureML -> Azure ML

File: cli/jobs/train/r/README.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+## Environment
+
+All the projects in this directory use the `rocker/tidyverse:4.0.0-ubuntu18.04` image from Docker Hub. This image has the tidyverse and its dependencies installed (see [Rocker](https://github.com/rocker-org/rocker) for more details).
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/README.md:17]: will this remain true over time (i.e. will we switch it to a curated image/environment we maintain?) - if so would remove

File: cli/jobs/train/r/README.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+In addition there are 2 things to point out about the [Dockerfile](./basic-train-model/Dockerfile) that builds the environment for these samples:
+
+In this section python is installed, since it is required by AzureML to start the job. This requirement will be removed for runs executed on cloud compute (AML Compute Cluster, AML Compute Instance) soon.
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/README.md:21]: don't use "AML" in documentation

File: cli/jobs/train/r/README.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+```Dockerfile
+RUN apt-get update -qq && \
+ apt-get install -y python3-pip
+RUN ln -f /usr/bin/python3 /usr/bin/python
+RUN ln -f /usr/bin/pip3 /usr/bin/pip
+RUN pip install -U pip
+```
+
+In this next section, the dependencies for local execution with data mounting are installed. This is required if you are using cloud data in your job and you are running the job on the compute target `local` (i.e. your local docker runtime).
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/README.md:30]: from my testing, you actually don't need these dependencies installed locally, but rather in the job's environment when running locally

File: cli/jobs/train/r/accident-prediction/accident_r_job.yml [hunk 14]
```diff
@@ -0,0 +1,20 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandJob.schema.json
+description: This job trains a GLM on the accidents dataset to predict fatality
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/accident-prediction/accident_r_job.yml:2]: I think the multiline description might be causing the issue with job creation

Comment (lostmygithubaccount) [inline cli/jobs/train/r/accident-prediction/accident_r_job.yml:2]: please follow order defined in contributing guide for sections of jobs, i.e. description toward the end

File: cli/jobs/train/r/accident-prediction/accident_r_job.yml [hunk 15]
```diff
@@ -0,0 +1,20 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandJob.schema.json
+description: This job trains a GLM on the accidents dataset to predict fatality 
+  based on features like airbag, seatbelt, etc.  
+code:
+  local_path: ./src
+command: >-
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/accident-prediction/accident_r_job.yml:6]: \>- -> > (no hyphen)

File: cli/jobs/train/r/accident-prediction/accident_r_job.yml (hunk 14 above)
Comment (lostmygithubaccount) [inline cli/jobs/train/r/accident-prediction/accident_r_job.yml:2]: match description with other jobs (see table)

File: cli/jobs/train/r/basic-train-model/basic_r_job.yml [hunk 16]
```diff
@@ -0,0 +1,17 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandJob.schema.json
+description: This job trains an rpart model on the iris dataset
```
Comment (lostmygithubaccount) [inline cli/jobs/train/r/basic-train-model/basic_r_job.yml:2]: match description with other jobs - follow order of job sections defined in contributing guidelines

**Review by lostmygithubaccount — CHANGES_REQUESTED** [review]

**Comment by lostmygithubaccount** [comment]
fyi: https://github.com/Azure/azure-cli-extensions/issues/3436
